dkg/dkg_test.go

```diff
@@ -36,25 +37,40 @@ import (
 	"github.com/obolnetwork/charon/dkg"
 	"github.com/obolnetwork/charon/p2p"
 	"github.com/obolnetwork/charon/tbls"
+	"github.com/obolnetwork/charon/tbls/tblsconv"
 	"github.com/obolnetwork/charon/testutil"
 	"github.com/obolnetwork/charon/testutil/keystore"
 )
 
 func TestDKG(t *testing.T) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
 	const (
 		nodes = 3
 		vals  = 2
 	)
 
+	t.Run("keycast", func(t *testing.T) {
+		lock, keys, _ := cluster.NewForT(t, vals, nodes, nodes, 0)
+		lock.Definition.DKGAlgorithm = "keycast"
+		testDKG(t, lock.Definition, keys)
+	})
+
+	t.Run("frost", func(t *testing.T) {
+		lock, keys, _ := cluster.NewForT(t, vals, nodes, nodes, 0)
+		lock.Definition.DKGAlgorithm = "frost"
+		testDKG(t, lock.Definition, keys)
+	})
+}
+
+func testDKG(t *testing.T, def cluster.Definition, p2pKeys []*ecdsa.PrivateKey) {
```

Comment:
nit: we can also have another test which checks the negative cases, like what happens when we have a malformed or incomplete definition/lock

Reply:
you are welcome to add these in another PR